src/internal/types.ts

```diff
@@ -259,3 +259,8 @@ export type ValueFromNotification<T> = T extends { kind: 'N' | 'E' | 'C' }
       : undefined
     : never
   : never;
+
+/**
+ * Used to help deal with rest arguments of many observable inputs that are tailed by schedulers, etc.
+ */
+export type OneOrMoreUnknownObservableInputs = readonly [ObservableInput<unknown>, ...unknown[]];
```

Comment:
issue: AFAICT, this makes things worse. The `...unknown[]` will match anything, so these don't effect errors:

```ts
const o5 = merge(a$, 1, 2, 3, 4, 5); // $ExpectError
const o6 = merge(a$, 1, 2, 3, 4, 5, asapScheduler); // $ExpectError
```

And, changing `...unknown[]` to `...ObservbleInput<unknown>[]` just makes things worse. I'll have another look at this after I've had a coffee, but ATM ... 🤷‍♂️

Reply:
I don't have a decent, actionable suggestion for this problem. I still think it's a TS bug. If it weren't for the concurrency parameter, we could keep the ugly overloads for the deprecated scheduler sigs and use variadic tuples for the non-deprecated sigs. But we have the trailing concurrency, so ... 🤷‍♂️

Reply:
Ugh... you're right. I tried `...ObservableInput<unknown>[]` _first_, but it didn't work and I couldn't see why.

@DanielRossenwasser, do you have any suggestions?

Reply:
Actually, I've fixed your example, @cartant, [here](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtVIB4BleEADwxFWAGd44pg8IBPeAbRoxi1QHMANPAB0orj37sAulIB8ACgBQ8EWO68+NAILUAcsgC2AIxAwAXPRCNmbdqOFEhqQyZhTFASgt3Rj+M+NTKQBuRUUwPC54KHgAXiR5ACIoRKEARg9QiNQooziE5NT4RKMijKzIjHgwfMR5cvDK+GBa+qEAJkz4AHpu+AB3DEQAfiA)

```ts
declare function f<S extends readonly [string, ...string[]]>(
  ...stringsAndNumber: readonly [...S, number]
): [...S, number];

const a = f("a", 1);
const b = f("a", "b", 1);
const c = f(1);
const d = f(1, 2); // works fine now (with an error)
```

Reply:
Oh... but it still doesn't work in our case, once there's a generic inside of that declaration... farts.

Reply:
See [here](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBC4AsA8AVAfNA3gKGtYA9gHYQAuATgK7CkHkAUADpQEYgCWw05ApmACbEQAT2hswCAFzQUASiwBfbEuz8eoML2gAzSkRrtiOpAGVoPAB6keRfjF4ChogNrwJSPQGsiBAO5E0ABpoADowt2QvH380ZwBdOLR6XFCwsnJ2IgBzCABBWwA5SgBbFh5yaQdBIhFoZzCQk2CiErLyOOxZaXqwpugW0vK4gG5sMcISUjFEAEZoAF5+nl84RHoAcjB12VGJsmmJACYFpZWI+nid8eJ9sBPtenEEGeCZq72plnvH2eCnw9e7xuUy4iweb12wOg-G+L2ghyBk3M3yecP+r2CAOgAGYdtAAPT46C+UjaAD8QA)

```ts
class Blah<T> {
  constructor(public readonly blah: T) {}
}

declare function f<S extends readonly [Blah<unknown>, ...Blah<unknown>[]]>(
  ...stringsAndNumber: readonly [...S, number]
): [...S, number];


const blah1 = new Blah('a');
const blah2 = new Blah([]);

const a = f(blah1, 1);
const b = f(blah1, blah2, 1);
const c = f(1);
const d = f(1, 2);
const e = f(blah1, blah2, 1, 2, 3); // wtf?
```

Reply:
Okay, I've pushed an update with a commit that _should work_ (I think), but doesn't. Marking this as blocked until we hear from the TS team about whether they're going to support that or not. It _looks_ like a simple bug, but maybe there's some other approach they want people to take.

Reply:
The TypeScript playground link that I added above was for what is, IMO, the fundamental bug.

The solution you have [above](https://github.com/ReactiveX/rxjs/pull/5844#discussion_r511007616) will work if the tuple of type parameters to `Blah` is extracted and mapped, like this:

```ts
class Blah<T> {
  constructor(public readonly blah: T) {}
}

type Blahs<T extends readonly unknown[]> = {
  [K in keyof T]: Blah<T[K]>;
};

declare function f<S extends readonly [unknown, ...unknown[]]>(
  ...stringsAndNumber: readonly [...Blahs<S>, number]
): [...Blahs<S>, number];
````

See this [playground snippet](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBC4AsA8AVAfNA3gKGtYA9gHYQAuATgK7CkHkAUADpQEYgCWw05ApmACbEQAT2hswCAFzQUASiwBfbEuylhjHnEQRU0HgA9SPIvxi8BQ0ZSIBrIgQDuRANoBdDAF4suaM4DS0OxE0DY8wgQAZjKu0vASqP7uANzKKdj8PKBgvNAR1jTsxLlIAMp6hsam3HyCRCK+1naORAA00AB0nY32Tm7u9D6d7WTkQQDmEACCJgBylAC2LDzk0ua19c5DcQg6JWhtRAtL5K7YstKbndu7+9CHi8uuadiEJKRiiACM0F5EPA5aCT0ADkYGBshSrzIHwkACYfnd-oCEPQ3BDsC9iNCwAiIvRxAhPm1Puioe8WLj8V82gTYcTSVj3lwvHiSZDGdB+JSidBYQy3npKQSebTiW06dAAMwQ6AAell0AcpAiAH4gA)

Whether that will work in the codebase, IDK. I'm pretty fed up with TS, TBH.